edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/util/PrometheusMetricsParser.cs

```
@@ -13,51 +13,135 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics.Util
+                                {
+                                    if (c == 'n')
+                                    {
+                                        result += "\n";
+                                    }
+                                    else
+                                    {
+                                        result += c;
+                                    }
+
+                                    escaped = false;
+                                }
+                                else if (c == '\\')
+                                {
+                                    escaped = true;
+                                }
+                                else
+                                {
+                                    result += c;
+                                }
+                            }
 
-                        string metricName = match.Groups["metricname"]?.Value ?? string.Empty;
-                        var tagNames = (match.Groups["tagname"].Captures as IEnumerable<Capture>).Select(c => c.Value);
-                        var tagValues = (match.Groups["tagvalue"].Captures as IEnumerable<Capture>).Select(c => c.Value);
+                            return result;
+                        });
 
-                        Dictionary<string, string> tags = tagNames.Zip(tagValues, (k, v) => new { k, v })
-                            .ToDictionary(x => x.k, x => x.v);
+                    Dictionary<string, string> tags = tagNames.Zip(tagValues, (k, v) => new { k, v })
+                        .ToDictionary(x => x.k, x => x.v);
+
+                    string metricTimestampString = match.Groups["metrictimestamp"]?.Value;
+                    if (metricTimestampString.Length != 0)
+                    {
+                        long metricTimestamp;
+                        if (!long.TryParse(metricTimestampString, out metricTimestamp))
+                        {
+                            continue;
```

Same comment as above about silent continue

---

Since the existing code already silently swallowed metrics that didn't match the regex at all, as well as metrics with bad metric values (as I explained in your previous comment), I figured it would make sense for this too.

---

I've added warnings to all three cases (regex doesn't match, metric value is malformed, metric timestamp is malformed).